lib/data/external/AwsClient.js

```diff
@@ -371,10 +371,12 @@ class AwsClient {
     objectPutTagging(key, bucket, objectMD, log, callback) {
         const awsBucket = this._awsBucketName;
         const awsKey = this._createAwsKey(bucket, key, this._bucketMatch);
-        const tagParams = { Bucket: awsBucket, Key: awsKey };
-        if (objectMD.versionId) {
-            tagParams.VersionId = objectMD.versionId;
-        }
+        const dataStoreVersionId = objectMD.location[0].dataStoreVersionId;
```

Comment:
Although I added the method to the class in an Arsenal PR, the object MD that is passed here is not an instance of the object MD class. It looks like so far we're only using the class when storing metadata, and haven't made changes yet to use it for metadata we've retrieved.

It seems a bit overkill to create the new instances in the AwsClient just to retrieve this property. I can still do it if you think it's better, or we can save it for the refactor (would be a large pr because the class makes properties private and accessible better only through the class getters/setters).

Reply:
ok. that's sad. skip it.